### PR TITLE
Issue 298 add more docs

### PR DESCRIPTION
**Pull request by dirmeier**
#### Submission Checklist

- [x] Builds locally 
- [x] Declare copyright holder and open-source license: see below

#### Summary

Ciao all, in case you're interested, this PR fixes some issues mentioned in #298, such as:

- adds docu for overloads of `size()` under functions reference 2.4 for scalars and 5.1 for matrices and vectors #120
- adds docu for Lambert functions under a new category "3.15 Special functions" in real-valued basic functions #200 
- moves docu of elementwise operations for matrices/vectors from 5.2.4 to 5.4 #149 
- adds docu for container construction under a new category "5.9 Container construction functions" #206
- adds a sentence to explain transpose #167

Issue #196 is already solved and can imo be closed.

Cheers,
Simon
 
#### Copyright and Licensing

Please list the copyright holder for the work you are submitting (this will be you or your assignee, such as a university or company): Simon Dirmeier

By submitting this pull request, the copyright holder is agreeing to license the submitted work under the following licenses:
- Code: BSD 3-clause (https://opensource.org/licenses/BSD-3-Clause)
- Documentation: CC-BY 4.0 (https://creativecommons.org/licenses/by/4.0/)

### CODE REVIEW

**Comment by rok-cesnovar** [comment]
Wow, this is very nice. Thank! I can review tomorrow if someone does not beat me too it.

File: src/functions-reference/matrix_operations.Rmd [hunk 1]
```diff
@@ -958,6 +976,112 @@ v)` (and similarly for pre-multiplication). By the same token, it is
...
+Create an identity matrix of size $k \times k$
+
+
+
+## Container construction functions {#container-construction}
+
+<!-- real[]; linspaced_array; (int n, real lower, real upper); -->
+\index{{\tt \bfseries linspaced\_array }!{\tt (int n, real lower, real upper): real[]}|hyperpage}
+
+`real[]` **`linspaced_array`**`(int n, real lower, real upper)`<br>\newline
```
Comment (rok-cesnovar) [inline src/functions-reference/matrix_operations.Rmd:992]:
```suggestion
`real[]` **`linspaced_array`**`(int n, data real lower, data real upper)`<br>\newline
```

File: src/functions-reference/matrix_operations.Rmd [hunk 2]
```diff
@@ -958,6 +976,112 @@ v)` (and similarly for pre-multiplication). By the same token, it is
...
+Create a real array of length `n` of equidistantly-spaced elements between `lower` and `upper`
+
+<!-- vector; linspaced_vector; (int n, real lower, real upper); -->
+\index{{\tt \bfseries linspaced\_vector }!{\tt (int n, real lower, real upper): vector}|hyperpage}
```
Comment (rok-cesnovar) [inline src/functions-reference/matrix_operations.Rmd:996]:
```suggestion
\index{{\tt \bfseries linspaced\_vector }!{\tt (int n, data real lower, data real upper): vector}|hyperpage}
```

File: src/functions-reference/matrix_operations.Rmd [hunk 3]
```diff
@@ -958,6 +976,112 @@ v)` (and similarly for pre-multiplication). By the same token, it is
...
+
+## Container construction functions {#container-construction}
+
+<!-- real[]; linspaced_array; (int n, real lower, real upper); -->
+\index{{\tt \bfseries linspaced\_array }!{\tt (int n, real lower, real upper): real[]}|hyperpage}
+
+`real[]` **`linspaced_array`**`(int n, real lower, real upper)`<br>\newline
+Create a real array of length `n` of equidistantly-spaced elements between `lower` and `upper`
+
+<!-- vector; linspaced_vector; (int n, real lower, real upper); -->
```
Comment (rok-cesnovar) [inline src/functions-reference/matrix_operations.Rmd:995]:
```suggestion
<!-- vector; linspaced_vector; (int n, data real lower, data real upper); -->
```

File: src/functions-reference/matrix_operations.Rmd [hunk 4]
```diff
@@ -958,6 +976,112 @@ v)` (and similarly for pre-multiplication). By the same token, it is
...
+\index{{\tt \bfseries linspaced\_vector }!{\tt (int n, real lower, real upper): vector}|hyperpage}
+
+`vector` **`linspaced_vector`**`(int n, real lower, real upper)`<br>\newline
```
Comment (rok-cesnovar) [inline src/functions-reference/matrix_operations.Rmd:998]:
```suggestion
`vector` **`linspaced_vector`**`(int n, data real lower, data real upper)`<br>\newline
```

File: src/functions-reference/matrix_operations.Rmd [hunk 5]
```diff
@@ -958,6 +976,112 @@ v)` (and similarly for pre-multiplication). By the same token, it is
...
+Create an `n`-dimensional vector of equidistantly-spaced elements between `lower` and `upper`
+
+<!-- row_vector; linspaced_row_vector; (int n, real lower, real upper); -->
```
Comment (rok-cesnovar) [inline src/functions-reference/matrix_operations.Rmd:1001]:
```suggestion
<!-- row_vector; linspaced_row_vector; (int n, data real lower, data real upper); -->
```

File: src/functions-reference/matrix_operations.Rmd [hunk 6]
```diff
@@ -958,6 +976,112 @@ v)` (and similarly for pre-multiplication). By the same token, it is
...
+\index{{\tt \bfseries linspaced\_row\_vector }!{\tt (int n, real lower, real upper): row\_vector}|hyperpage}
```
Comment (rok-cesnovar) [inline src/functions-reference/matrix_operations.Rmd:1002]:
```suggestion
\index{{\tt \bfseries linspaced\_row\_vector }!{\tt (int n, data real lower, data real upper): row\_vector}|hyperpage}
```

File: src/functions-reference/matrix_operations.Rmd [hunk 7]
```diff
@@ -958,6 +976,112 @@ v)` (and similarly for pre-multiplication). By the same token, it is
...
+
+`row_vector` **`linspaced_row_vector`**`(int n, real lower, real upper)`<br>\newline
```
Comment (rok-cesnovar) [inline src/functions-reference/matrix_operations.Rmd:1004]:
```suggestion
`row_vector` **`linspaced_row_vector`**`(int n, data real lower, data real upper)`<br>\newline
```

**Review by rok-cesnovar — CHANGES_REQUESTED** [review]
Thank you very much for this! 

Arguments of type real in "Container construction functions" should have the data qualifier. Other than that it looks good to me.

File: src/functions-reference/matrix_operations.Rmd (hunk 7 above)
Comment (dirmeier) [reply src/functions-reference/matrix_operations.Rmd]: Hey @rok-cesnovar , thanks for the quick review. All done.
Cheers,
Simon

**Review by rok-cesnovar — APPROVED** [review]

**Pull request merged**